include/core/simulation_control.h

```diff
@@ -346,24 +349,13 @@ class SimulationControl
   void
   set_current_time_step(const double new_time_step)
   {
+    Assert(
```

Comment:
I don't see a situation where a time_step of 0 would be used. I would change to time_step > 0.